src/transformers/trainer.py

```diff
@@ -3307,6 +3307,11 @@ def training_step(self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor,
             loss = self.compute_loss(model, inputs)
 
         del inputs
+        if (
+            self.args.torch_empty_cache_steps is not None
+            and self.state.global_step % self.args.torch_empty_cache_steps == 0
+        ):
+            torch.cuda.empty_cache()
```

Comment:
So after looking at the accelerate issue about this, we actually need to be *very* careful since we are device agnostic. 

You can do something like this:
```python
    if is_xpu_available():
        torch.xpu.empty_cache()
    elif is_mlu_available():
        torch.mlu.empty_cache()
    elif is_npu_available():
        torch.npu.empty_cache()
    elif is_mps_available(min_version="2.0"):
        torch.mps.empty_cache()
    else:
        torch.cuda.empty_cache()
```

Reply:
This would mean I need to import `accelerate` which makes it no longer an optional dep...unless we want to make this argument to only work with accelerate enabled. Is that ideal?

Reply:
If we’re using PyTorch, it’s always mandatory and has been for almost a year now. 

However use this snippet so we don’t need to rely on the latest version.

Reply:
`is_mps_available()` dont have a `min_version` and its implemented as:
```python
def is_mps_available():
    return is_torch_version(">=", "1.12") and torch.backends.mps.is_available() and torch.backends.mps.is_built()
```

Reply:
We don’t need a min version, just that exact chunk of code I pasted there should be used instead of just CUDA.empty_cache()

Reply:
You can combine the transformers is mps available + a check that the torch version is about 2.0

Reply:
Pushed a commit